mesh/meshdb.go

```diff
@@ -362,6 +363,18 @@ func getRewardKeyPrefix(account types.Address) []byte {
 	return []byte(str)
 }
 
+// adding functions for getting the reward key for a smesherID
+// format for the index "reward_<smesherid>_<layerid> -> reward_<accountid>_<layerid> -> the actual reward"
+func getSmesherRewardKey(l types.LayerID, smesherID types.NodeID) []byte {
+	str := string(getSmesherRewardKeyPrefix(smesherID)) + "_" + strconv.FormatUint(l.Uint64(), 10)
+	return []byte(str)
+}
+
+func getSmesherRewardKeyPrefix(smesherID types.NodeID) []byte {
+	str := "reward_" + smesherID.String()
```

Comment:
Probably more efficient to use `binary.BigEndian.PutUint*` here as well

Also, this seems to conflict with what `getRewardKeyPrefix` does, above - do we need two separate methods?

Reply:
I added two separate methods so that we could do a prefix lookup on the SmesherRewardKey (which is a secondary key in this abstraction) to obtain all the rewards for a given Smesher